sdk/src/main/java/com/mapbox/maps/MapController.kt

```diff
@@ -125,12 +132,21 @@ internal class MapController : MapPluginProviderDelegate, MapControllable {
   }
 
   override fun onDestroy() {
+    if (lifecycleState == LifecycleState.STATE_DESTROYED) {
+      return
+    }
+    lifecycleState = LifecycleState.STATE_DESTROYED
+
     mapboxMap.onDestroy()
     nativeObserver.clearListeners()
     renderer.onDestroy()
     pluginRegistry.cleanup()
   }
 
+  override fun onLowMemory() {
+    mapboxMap.reduceMemoryUse()
+  }
```

Comment:
Shall we add some state checks similar to onStart etc, to avoid calling mapboxMap.reduceMemoryUse() multiple times?

Reply:
No, this is not in the regular lifecycle and could be invoked multi times.